doc/reference/glossary.rst

```diff
@@ -29,6 +29,44 @@ Glossary
       all nodes). It can be a list, set, graph, etc.. To filter an nbunch
       so that only nodes actually in `G` appear, use `G.nbunch_iter(nbunch)`.
 
+      If the nbunch is a container or iterable that is not itself a node
+      in the graph, then it will be treated as an iterable of nodes, for
+      instance, when nbunch is a string or a tuple.
+
+      .. nbplot::
+
+         >>> import networkx as nx
+         >>> G = nx.DiGraph()
+         >>> G.add_edges_from([("b", "c"), ("a", "ab"), ("ab", "c")])
+         >>> G.edges("ab")
+         OutEdgeDataView([('ab', 'c')])
+      
+      Since "ab" is a node in G, it is treated as a single node.
+
+      .. nbplot::
+         
+         >>> G.edges("bc")
+         OutEdgeDataView([('b', 'c')])
+
+      Since "bc" is not a node in G, it is treated as an iterator.
+
+      .. nbplot::
+         
+         >>> G.edges(["bc"])
+         OutEdgeDataView([])
+
+      If "bc" is wrapped in a list, the list is the iterable and
+      "bc" is treated as a single node.
```

Comment:
This could be reworded to frame it as a list of iterables. Which is more clear? The current wording focuses on a single example (from which the reader can abstract to the general case). The wording below focuses on the general case, but maybe it then becomes too abstract.  Which do you think is better? I'm not sure...

```
    If the nbunch is an iterable of iterables, the inner iterables will always be treated as nodes. So, a list of strings ensures that the string is treated as a node and not treated as an iterable of nodes.
```

Maybe the best is to combine both into a single paragraph. And leaving it as it is would work too. The decision is yours.

Reply:
>  If the nbunch is an iterable of iterables, the inner iterables will always be treated as nodes.

I think this sentence should definitely be added! Else someone might find it difficult to generalize `list of strings` to other use cases, like a tuple of tuples.